grpc/execution/validation.go

```diff
@@ -0,0 +1,63 @@
+package execution
+
+import (
+	sequencerblockv1alpha1 "buf.build/gen/go/astria/sequencerblock-apis/protocolbuffers/go/astria/sequencerblock/v1alpha1"
+	"crypto/sha256"
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// if sequencer tx is valid, then a unmarshalled ethereum transaction is returned. if not valid, nil is returned
+func (s *ExecutionServiceServerV1Alpha2) ValidateAndUnmarshalSequencerTx(tx *sequencerblockv1alpha1.RollupData) (*types.Transaction, error) {
```

Comment:
I'm not sure this is best as a method on the execution server. Maybe best choice for now, but only thing it's ingesting is bridge config information. Perhaps it could just take those as parameters?

This also only validates deposit tx types. Perhaps those should be validated elsewhere similar to how we validate the eth txs elsewhere just build the deposit don't validate validate elsewhere? These could be follow up changes though

Reply:
yeah i thought the same too. I think it makes sense to pass the bridge config

Reply:
yeah i think we can refactor the validation logic in the future.